### DISCUSSION TITLE
Apply turbulence at the entrance of a ring

### DISCUSSION THREAD

**Discussion by hzw-zju-2023** [AMReX-Combustion/PeleC#816] [Q&A]
I am simulating swirl combustion. The swirl inlet is a ring, and I want to apply turbulence at the entrance of this ring. What I do is to generate homogeneous isotropic turbulence box with reference to TurbFileHIT/ case, and then specify related parameters in the input file. My current speed results are shown in the figure. Turbulence doesn't seem to be developing very well.
One problem with turb_scale_loc is that I don't know how to determine it. Since the ring is composed of a big circle and a small circle, I adopted my previous case of circular tube jet, turb_scale_loc was obtained by dividing 2pi by the diameter of the big circle. But I don't know if it's a struggle. One problem with turb_scale_loc is that I don't know how to determine it. Since the ring is composed of a big circle and a small circle, I adopted my previous case of circular tube jet, turb_scale_loc was obtained by dividing 2pi by the diameter of the big circle. But I'm not sure it's the right thing to do. I would like to ask if there is a better rational way to apply turbulence at the entrance of a ring?

**Comment by hzw-zju-2023** [comment]
![图片1 jpg](https://github.com/AMReX-Combustion/PeleC/assets/154048455/dfc7d22f-36c1-432d-808d-d5558b36271f)
![图片2 jpg](https://github.com/AMReX-Combustion/PeleC/assets/154048455/43cf796b-438c-439f-b0fd-f1eff75a00de)

**Reply by baperry2** [reply]
If I understand correctly, your approach is generally correct. This method of adding turbulence is ad hoc, and mostly intended for situations where having some velocity fluctuations is necessary to activate instabilities in the main flow, but a high-fidelity representation of the incoming turbulence is not required. Obviously, applying isotropic turbulence on an annular mean flow will not accurately represent the incoming turbulence. You can try to adjust the turbinflow parameters and the parameters in the HIT generation script to achieve incoming turbulence that you find suitable. Given the ad hoc nature of this approach, there isn't really a rigorous way to determine the parameters and some trial and error will likely be needed.

If you want higher fidelity incoming turbulence, you would need to simulate the upstream flow with Pele or another CFD solver and figure out how to save data in the format expected by the turbinflow utility. I do not believe there are existing tools for this purpose.

**Reply by drummerdoc** [reply]
The turbInflow stuff in fact was originally written to support this mode (storing off time-dependent planes to shove into a new sim), but that option was not carried along in recent versions of the code because nobody was using it.  The reality is that it is only a small modification to the file format, and to the interpolation (avoiding periodic wrap-around in the flow direction, and tracking the time-stamp of each plane and ensuring that the parabolic interpolant respects the specific time spacing (rather than assuming it to be uniform in space/time, as is done with the periodic inflow code).  If it is determined to be useful, and if we had a test case to ensure correctness, I think we can add back the code to do this with little effort. (that is, if you decide that you want to precompute the time-dependent swirling turbulent inflow and shove that into your PeleC/LMeX calc, and are willing to manage the data files to ensure that the BC stuff doesn't try to access time planes that don't exist in the input, let me know and I can work with you to get that functioning).

**Reply by CAIROCharlie** [reply]
@drummerdoc Thank you for the detailed explanation! It's also very inspiring to me. This functionality sounds very interesting and potentially quite useful for my current work. I am currently looking to input turbulent flow data from a circular pipe into my simulation as an inflow boundary condition. If possible, I would greatly appreciate your guidance in helping me implement this. Thanks again.

**Reply by drummerdoc** [reply]
Sure, I can help.  Again, the most annoying part of this is dealing with the non-periodic data in the flow direction, which requires you to either precompute the entire flow field ahead of time and deal with a massive input file, or work in sections and always ensure that the currently active section spans the time period where you need inflow data....  Reach out to me at Marc.Day@nrel.gov next wee and we can talk more.
